fml/synchronization/critical_section.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_SYNCHRONIZATION_CRITICAL_SECTION_H_
+#define FLUTTER_FML_SYNCHRONIZATION_CRITICAL_SECTION_H_
+
+#include <mutex>
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/memory/ref_counted.h"
+
+namespace fml {
+
+class CriticalSection : public fml::RefCountedThreadSafe<CriticalSection> {
```

Comment:
Isn't this just a `scoped_lock` or `lock_guard` around a `std::shared_ptr<std::mutex>`?

Reply:
Yes, I didn't find a suitable lock at the beginning, so I added one.  :)  It may be redundant.

If you guys think this solution is effective, I will refactor the code and improve the test case. thanks.